runtimes/alibi-explain/mlserver_alibi_explain/common.py

```diff
@@ -82,17 +101,49 @@ def import_and_get_class(class_path: str) -> type:
     return klass
 
 
-def to_v2_inference_request(input_data: Union[np.ndarray, List[str]]):
+def to_v2_inference_request(
+    input_data: Union[np.ndarray, List[str]],
+    metadata: Optional[MetadataModelResponse],
+) -> InferenceRequest:
+    """
+    Encode numpy payload to v2 protocol.
+
+    Note: We only fetch the first-input name and the list of outputs from the metadata
+    endpoint currently. We should consider wider reconciliation with data types etc.
+
+    Parameters
+    ----------
+    input_data
+       Numpy ndarray to encode
+    metadata
+       Extra metadata that can help encode the payload.
+    """
+
+    # MLServer does not really care about a correct input name!
+    input_name = _DEFAULT_INPUT_NAME
+    id_name = _DEFAULT_ID_NAME
+    outputs = []
+
+    if metadata is not None:
+        if metadata.inputs:
+            # we only support a big single input numpy
+            input_name = metadata.inputs[0].name
+        if metadata.outputs:
+            outputs = metadata.outputs
+
     # For List[str] (e.g. AnchorText), we use StringCodec for input
     input_payload_codec = StringCodec if type(input_data) == list else NumpyCodec
     v2_request = InferenceRequest(
+        id=id_name,
```

Comment:
I'm not 100% sure, but I think the executor will set this to a unique value. Therefore, should we leave the `id` field empty?

Reply:
I am not sure about SC, maybe this is the case. However locally it will not work with a None id (for triron case).